### PR TITLE
✨ Introduce API study endpoints for `start` `stop` `delete` job

### PR DESCRIPTION
**Pull request by bisgaard-itis**
## What do these changes do?
- Introduce `start`, `stop` and `delete` endpoints in the api-server for study jobs.
- @wvangeit I did not add any tests for the `create_study_job` endpoint in this PR, but I needed the implementation to be complete, so I did an attempt at that. Probably it makes sense that you merge your implementation first though, which might require me to perform some changes here.

## Related issue/s

- Implements the 3 first endpoints in the following list: https://github.com/ITISFoundation/osparc-issues/issues/1291

## How to test

- Run the api-server unit tests (`cd services/api-server/tests/unit; python -m pytest .`).
## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: packages/models-library/src/models_library/api_schemas_webserver/projects_ports.py
Comment (pcrespov) [inline packages/models-library/src/models_library/api_schemas_webserver/projects_ports.py:7]: Eventually these configs should be the same as those in `packages/models-library/src/models_library/api_schemas_webserver/_base.py`.  
Unfortunately the web-api is not consistent enough for this  but could you please check if you can use the ones in _base? Otherwise it is ok

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:99]: TIP: the catalog of some non-containerized services are in `models_library.function_services_catalog.services`. For file picker you can
```python
from models_library.function_services_catalog.services import file_picker

# ...
if (node.kye == file_picker.META.key and len(node.outputs) == 0):
     # etc

```

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:210]: TIP: we could use `log_context` for all these?

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py:37]: Fix fixmes

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py:70]: same here

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py:110]: `return Job(...)`

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:264]: why not simply 
```python
return await nodes_utils.update_node_outputs(
        app=request.app,
        user_id=req_ctx.user_id,
        project_uuid=path_params.project_id,
        node_uuid=path_params.node_id,
        outputs=node_outputs.outputs,
        run_hash=None,
        node_errors=None,
        ui_changed_keys= { path_params.node_id },
)
```

**Review by pcrespov — APPROVED** [review]
Nice.
Left some suggestions.

File: services/api-server/src/simcore_service_api_server/utils/http_calls_capture_processing.py
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/utils/http_calls_capture_processing.py:58]:
```python

if not any([type_, oneOf, anyOf, allOf]):
    type_ = values["type_"] = "str"

# Validate that 'pattern' and 'format_' are None for types other than "str"
if type_ != "str" and any([pattern, format_]):
    raise ValueError(f"Type {type_} requires both pattern and format to be None")
```

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py
Comment (matusdrobuliak66) [inline packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py:125]: Minor: If there is some structure to this dictionary it would be nice to use type hints

**Review by sanderegg — APPROVED** [review]
thanks! just a few comments and a TODO to fix/change/remove?

**Comment by bisgaard-itis** [comment]
> Thx a lot @bisgaard-itis , looks very good. I would suggest we just start with your implementation of create_study_job and go from there. I can see if I need something more to get my use case working, and transfer some of my code if necessary.

@wvangeit hmm, I think it would be better to try and ensure that the implementation is what we want first instead of merging and then figuring out what we want, but we can do like that if you prefer. In any case we will need a test for it. Do you already have one?

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (GitHK) [inline services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:97]: how are you sure this is always the case? I would drag this in the if condition or raise and handle an error in that situation

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (GitHK) [inline services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py:105]: there is a todo left here

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:99]: Thanks a lot for pointing this out Pedro. I completely forgot about this. Before going on holidays I wanted to correct this, but didn't manage to do it and forgot. I guess we currently don't have a better way to check that a node is a given service, but perhaps it would make sense to create such functionality 🤔

Comment (wvangeit) [reply services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:99]: @pcrespov yeah, this was my first naive implementation of this.

Another option we could consider, is letting the webserver return empty filepickers as inputs. But that might change some functionality. Or adding a 'file parameter'.

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py
Comment (wvangeit) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:247]: This is something I added here in my code. I naively used node.create. But since we're not really creating, but changing, it didn't seem appropriate?

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py
Comment (wvangeit) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:264]: That's just my coding style. I prefer to explicitly name things to make things more readable. But I'm ok with either.

**Review by YuryHrytsuk — APPROVED** [review]

**Comment by bisgaard-itis** [comment]
> > Thx a lot @bisgaard-itis , looks very good. I would suggest we just start with your implementation of create_study_job and go from there. I can see if I need something more to get my use case working, and transfer some of my code if necessary.
> 
> @wvangeit hmm, I think it would be better to try and ensure that the implementation is what we want first instead of merging and then figuring out what we want, but we can do like that if you prefer. In any case we will need a test for it. Do you already have one?

So how is this looking @wvangeit? Will you merge your implementation with some tests first? Because then I wait for that. If not I will leave out the creation endpoint in this PR and focus on getting the other three endpoints I committed to getting in. I don't want to merge some half-baked stuff in this one.

**Comment by wvangeit** [comment]
@bisgaard-itis i would say it's best you go ahead and merge in your endpoints. After that merge I might maybe have more success with my tests, but let's not get blocked by me.

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py:211]: Yep, makes sense. I will adapt. Thanks

File: services/api-server/src/simcore_service_api_server/utils/http_calls_capture_processing.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/utils/http_calls_capture_processing.py:58]: Thanks, this is definitely cleaner. I will adapt.

File: packages/models-library/src/models_library/api_schemas_webserver/projects_ports.py
Comment (bisgaard-itis) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_ports.py]: I removed this code for another PR, hence resolving this conversation

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py]: I removed this code for another PR, hence resolving this conversation

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py]: I removed this code for another PR, hence resolving this conversation

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py]: I removed this code for another PR, hence resolving this conversation

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py
Comment (bisgaard-itis) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py]: I removed this code for another PR, hence resolving this conversation

File: packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py
Comment (bisgaard-itis) [reply packages/models-library/src/models_library/api_schemas_webserver/projects_nodes.py]: I removed this code for another PR, hence resolving this conversation

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py
Comment (bisgaard-itis) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py]: I removed this code for another PR, hence resolving this conversation

File: services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/api/routes/studies_jobs.py]: I removed this code for another PR, hence resolving this conversation

File: services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py
Comment (bisgaard-itis) [reply services/api-server/src/simcore_service_api_server/services/study_job_models_converters.py]: I removed this code for another PR, hence resolving this conversation

**Comment by bisgaard-itis** [comment]
@pcrespov I rerequested a review from you just so you know what will be merged here. Up to fixing tests (in case some are failing now) this should be ready to merge from my side

**Review by pcrespov — APPROVED** [review]

**Review by wvangeit — APPROVED** [review]

**Pull request merged**
